benchmark/distinct_user_id/rapidjson_sax.h

```diff
@@ -0,0 +1,59 @@
+#pragma once
+
+#ifdef SIMDJSON_COMPETITION_RAPIDJSON
+
+#include "distinct_user_id.h"
+#include <string.h>
+namespace distinct_user_id {
+
+using namespace rapidjson;
+
+struct rapidjson_sax {
+    struct Handler {
+        std::vector<uint64_t>* result;
+        bool user = false;
+        bool user_id =false;
+        Handler(std::vector<uint64_t> &r) { result=&r; }
+
+        bool Key(const char* key, SizeType length, bool copy) {
+            if (strcmp(key,"user") == 0) { user = true; } // Checking if entering user object
```

Comment:
This code will do two string comparisons per key which is too much. Consider reducing this by doing something like...

```
  if (user) { look for id }
  else {look for user} 
```

Or something of the sort.